File: server/module/module-erp/erp-server/src/convert/erp_financial_record.rs
use sea_orm::{Set, NotSet};
use crate::model::erp_financial_record::{self, Model as ErpFinancialRecord, ActiveModel as ErpFinancialRecordActiveModel};
use erp_model::request::erp_financial_record::{CreateErpFinancialRecordRequest, UpdateErpFinancialRecordRequest};
use erp_model::response::erp_financial_record::ErpFinancialRecordResponse;

pub fn create_request_to_model(request: &CreateErpFinancialRecordRequest) -> ErpFinancialRecordActiveModel {
    ErpFinancialRecordActiveModel {
        record_type: Set(request.record_type.clone()),
        amount: Set(request.amount.clone()),
        description: request.description.as_ref().map_or(NotSet, |description| Set(Some(description.clone()))),
        related_order_id: request.related_order_id.as_ref().map_or(NotSet, |related_order_id| Set(Some(related_order_id.clone()))),
        record_date: Set(request.record_date.clone()),
        user_id: request.user_id.as_ref().map_or(NotSet, |user_id| Set(Some(user_id.clone()))),
        department_code: Set(request.department_code.clone()),
        department_id: Set(request.department_id.clone()),
        ..Default::default()
    }
}

pub fn update_request_to_model(request: &UpdateErpFinancialRecordRequest, existing: ErpFinancialRecord) -> ErpFinancialRecordActiveModel {
    let mut active_model: ErpFinancialRecordActiveModel = existing.into();
    if let Some(record_type) = &request.record_type { 
        active_model.record_type = Set(record_type.clone());
    }
    if let Some(amount) = &request.amount { 
        active_model.amount = Set(amount.clone());
    }
    if let Some(description) = &request.description { 
        active_model.description = Set(Some(description.clone()));
    }
    if let Some(related_order_id) = &request.related_order_id { 
        active_model.related_order_id = Set(Some(related_order_id.clone()));
    }
    if let Some(record_date) = &request.record_date { 
        active_model.record_date = Set(record_date.clone());
    }
    if let Some(user_id) = &request.user_id { 
        active_model.user_id = Set(Some(user_id.clone()));
    }
    if let Some(department_code) = &request.department_code { 
        active_model.department_code = Set(department_code.clone());
    }
    if let Some(department_id) = &request.department_id { 
        active_model.department_id = Set(department_id.clone());
    }
    active_model
}

pub fn model_to_response(model: ErpFinancialRecord) -> ErpFinancialRecordResponse {
    ErpFinancialRecordResponse { 
        id: model.id,
        record_type: model.record_type,
        amount: model.amount,
        description: model.description,
        related_order_id: model.related_order_id,
        record_date: model.record_date,
        user_id: model.user_id,
        department_code: model.department_code,
        department_id: model.department_id,
        creator: model.creator,
        create_time: model.create_time,
        updater: model.updater,
        update_time: model.update_time,
    }
}

File: admin/src/api/erp_financial_record.ts
import { PaginatedRequest, PaginatedResponse } from '@/base/page';
import { api } from '@/utils/request';

const apis = {
  create: '/erp_financial_record/create', // 新增
  update: '/erp_financial_record/update', // 修改
  delete: '/erp_financial_record/delete', // 删除
  get: '/erp_financial_record/get', // 单条查询
  list: '/erp_financial_record/list', // 列表查询
  page: '/erp_financial_record/page', // 分页查询
}

export interface ErpFinancialRecordRequest {
  id: number; // 财务记录ID
  record_type: number; // 记录类型 (0=income, 1=expense)
  amount: number; // 金额
  description: string; // 描述
  related_order_id: number; // 关联订单ID
  record_date: string; // 记录日期
  user_id: number; // 用户ID
  department_code: string; // 部门编码
  department_id: number; // 部门ID
}

export interface ErpFinancialRecordResponse {
  id: number; // 财务记录ID
  record_type: number; // 记录类型 (0=income, 1=expense)
  amount: number; // 金额
  description: string; // 描述
  related_order_id: number; // 关联订单ID
  record_date: string; // 记录日期
  user_id: number; // 用户ID
  department_code: string; // 部门编码
  department_id: number; // 部门ID
  creator: number; // 创建者ID
  create_time: string; // 创建时间
  updater: number; // 更新者ID
  update_time: string; // 更新时间
}

export interface ErpFinancialRecordQueryCondition extends PaginatedRequest {

}

export const createErpFinancialRecord = (erp_financial_record: ErpFinancialRecordRequest): Promise<number> => {
  return api.post<number>(apis.create, erp_financial_record);
}

export const updateErpFinancialRecord = (erp_financial_record: ErpFinancialRecordRequest): Promise<void> => {
  return api.post<void>(apis.update, erp_financial_record);
}

export const deleteErpFinancialRecord = (id: number): Promise<void> => {
  return api.post<void>(`${apis.delete}/${id}`);
}

export const getErpFinancialRecord = (id: number): Promise<ErpFinancialRecordResponse> => {
  return api.get<ErpFinancialRecordResponse>(`${apis.get}/${id}`);
}

export const listErpFinancialRecord = (): Promise<Array<ErpFinancialRecordResponse>> => {
  return api.get<Array<ErpFinancialRecordResponse>>(apis.list);
}

export const pageErpFinancialRecord = (condition: ErpFinancialRecordQueryCondition): Promise<PaginatedResponse<ErpFinancialRecordResponse>> => {
  return api.get<PaginatedResponse<ErpFinancialRecordResponse>>(apis.page, condition);
}
